### PR TITLE
Contributing guidelines

### PR DESCRIPTION
**Pull request by vblagoje**
For a preview see https://github.com/vblagoje/infinispan/blob/t_contributing/README.md Also verify that https://github.com/vblagoje/infinispan/blob/t_contributing/CONTRIBUTING.md and https://github.com/vblagoje/infinispan/blob/t_contributing/dco.txt

### CODE REVIEW

**Comment by tristantarrant** [comment]
@richardfontana can you have a look please ?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+Contributions from the community are essential in keeping Infinispan strong and successful.
+
+This guide focuses on how to contribute back to Infinispan using GitHub pull requests.
+If you need help with cloning, compiling or setting the project up in an IDE please refer to
+[this page](http://infinispan.org/docs/stable/contributing/contributing.html).
+
+## Legal
+
+All original contributions to Infinispan are licensed under the
+[ASL - Apache License](https://www.apache.org/licenses/LICENSE-2.0),
```
Comment (rvansa) [inline CONTRIBUTING.md]: link is to 2.0, but you say 2.1+

**Comment by rvansa** [comment]
Does this mean that we need to sign-off all commits since now? (Good that this [can be configured ](https://jjasghar.github.io/blog/2016/10/04/signing-commits-in-git/). Will there be any enforcement on GitHub so that as a reviewer I don't need to check that?

**Comment by richardfontana** [comment]
@tristantarrant It looks good except as @rvansa points out ASL is 2.0, not 2.1.

**Comment by richardfontana** [comment]
@rvansa speaking for Red Hat Legal that would be our strong preference.

**Comment by tristantarrant** [comment]
Fixed and merged.

**Pull request closed without merging**